### PR TITLE
executor: use MVMap for aggregate

### PR DESCRIPTION
**Pull request by coocood**
To reduce memory allocation and GC overhead.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @hanfei1991 PTAL

File: util/mvmap/mvmap.go [hunk 1]
```diff
@@ -85,6 +85,15 @@ func (ds *dataStore) get(e entry, key []byte) []byte {
 	return slice[valOffset : valOffset+e.valLen]
 }
 
+func (ds *dataStore) getEntryData(e entry) (key, value []byte) {
+	slice := ds.slices[e.addr.sliceIdx]
```
Comment (zhexuany) [inline util/mvmap/mvmap.go]: if `sliceIdx` already target than 0, then it is safe to do so. Is that possible `sliceIdx` being negative?

File: util/mvmap/mvmap.go [hunk 2]
```diff
@@ -170,3 +181,42 @@ func (m *MVMap) hash(key []byte) uint64 {
...
+
+// Iterator is used to iterate the MVMap.
+type Iterator struct {
+	m        *MVMap
+	sliceCur int
+	entryCur int
+}
+
+// Next return the next key/value pair of the MVMap.
+// It returns nil, nil when there is no more entry to iterate.
```
Comment (zhexuany) [inline util/mvmap/mvmap.go]: may be (nil, nil) is better?

File: util/mvmap/mvmap.go [hunk 3]
```diff
@@ -170,3 +181,42 @@ func (m *MVMap) hash(key []byte) uint64 {
...
+func (i *Iterator) Next() (key, value []byte) {
+	for {
+		if i.sliceCur >= len(i.m.entryStore.slices) {
+			return nil, nil
+		}
+		entrySlice := i.m.entryStore.slices[i.sliceCur]
+		if i.entryCur >= len(entrySlice) {
+			i.sliceCur++
+			i.entryCur = 0
```
Comment (zhexuany) [inline util/mvmap/mvmap.go]: When you creating a new `*Iterator` via `NewIterator`, you initialize `entryCur` as 1. But here you reassign 0 to `entryCur`. If this is a new initialization, I believe `1` is the answer. I may be wrong and it is highly possible, but it is worth of indicating this suspicious code.

File: util/mvmap/mvmap.go (hunk 1 above)
Comment (coocood) [inline util/mvmap/mvmap.go]: the sliceIdx is managed by MVMap, it will never be negative.

File: util/mvmap/mvmap.go (hunk 3 above)
Comment (coocood) [inline util/mvmap/mvmap.go]: Only the first entry in the first slice represents null, then it starts from 1, other slice starts from 0.

**Comment by shenli** [comment]
@tiancaiamao @zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**